### PR TITLE
Update to gradle 3.0

### PR DESCRIPTION
**Pull request by passsy**

### CODE REVIEW

**Comment by StefMa** [comment]
Do not merge since bintray-release can't handle this and we were unable to make a release with that :)

File: thirtyinch-rx/build.gradle [hunk 1]
```diff
@@ -36,14 +33,15 @@ android {
 }
 
 dependencies {
-    compile project(':thirtyinch')
-    compile 'io.reactivex:rxjava:1.3.2'
-    compile 'com.artemzin.rxjava:proguard-rules:1.2.7.0'
-    provided "com.android.support:support-annotations:$supportLibraryVersion"
+    implementation project(':thirtyinch')
+    implementation 'io.reactivex:rxjava:1.3.2'
```
Comment (StefMa) [inline thirtyinch-rx/build.gradle]: use `api`

File: thirtyinch-rx2/build.gradle [hunk 2]
```diff
@@ -36,13 +33,13 @@ android {
 }
 
 dependencies {
-    compile project(':thirtyinch')
-    compile 'io.reactivex.rxjava2:rxjava:2.1.4'
-    provided "com.android.support:support-annotations:$supportLibraryVersion"
+    implementation project(':thirtyinch')
+    implementation 'io.reactivex.rxjava2:rxjava:2.1.4'
```
Comment (StefMa) [inline thirtyinch-rx2/build.gradle]: use `api`

**Comment by passsy** [comment]
@StefMa wanny try https://github.com/novoda/bintray-release/releases/tag/0.8.0?

**Comment by StefMa** [comment]
For testing I've updloaded it to my personal bintray account.
Use
```
repositories {
    maven { url "https://dl.bintray.com/stefma/maven" }
}

dependencies {
    def thirtyinchVersion = '0.8.6-TEST'
    implementation "net.grandcentrix.thirtyinch:thirtyinch:$thirtyinchVersion"
    implementation "net.grandcentrix.thirtyinch:thirtyinch-rx2:$thirtyinchVersion"
    implementation "net.grandcentrix.thirtyinch:thirtyinch-logginginterceptor:$thirtyinchVersion"
}
```

**Comment by StefMa** [comment]
Hmm somehow the Rx2 module wasn't uploaded successfully 🤔 
Strange.
-> https://dl.bintray.com/stefma/maven/net/grandcentrix/thirtyinch/thirtyinch-rx2/0.8.6-TEST/

**Comment by StefMa** [comment]
I have upload it again and now it works... Strange 😱

**Comment by StefMa** [comment]
Seems to be a "one time issue".
No issues with https://dl.bintray.com/stefma/maven/net/grandcentrix/thirtyinch/thirtyinch-rx2/0.8.7-TEST/ so far 👍

**Comment by StefMa** [comment]
Checked `./gradlew app:dependencies`.
Seems that everything got packed correctly:
bildschirmfoto 2017-11-30 um 16 32 13
ThirtyInch brings no dependencies with in..
But Rx2-module brings Rx-Java2 but updated it internally to 2.1.5 (because the app themself depends on version 2.1.5) 🎉 

Seems that everything is correct 👍

**Comment by passsy** [comment]
Verified, all dependencies can be resolved correctly. I was quite surprised our core module has no external dependencies 😮

**Pull request merged**